netty/src/main/java/io/grpc/netty/NettyClientTransport.java

```diff
@@ -194,6 +199,27 @@ public ClientStream newStream(
     if (channel == null) {
       return new FailingClientStream(statusExplainingWhyTheChannelIsNull, tracers);
     }
+    try {
+      if (callOptions.getAuthority() != null && !negotiator.mayBeVerifyAuthority(
+          callOptions.getAuthority())) {
+        String errMsg = String.format("Peer hostname verification failed for authority '%s'.",
```

Comment:
Nit: we don't include ending punctuation in status error messages.

Reply:
Done.